src/modules/authentification/components/AuthRegisterForm.tsx

```diff
@@ -35,81 +50,123 @@ export default function AuthRegisterForm() {
-          <InputField
-            name="password"
-            placeholder="Enter a password..."
-            type="password"
-            validation={{
-              required: true,
-              minLength: {
-                value: passwordMinLength,
-                message: `Password must be atleast ${passwordMinLength} characters long.`,
-              },
-              maxLength: {
-                value: passwordMaxLength,
-                message: `Password cannot be longer than ${passwordMaxLength} characters.`,
-              },
-              pattern: {
-                value: passwordRegexp,
-                message:
-                  "Password must contain atleast one uppercase letter, one lowercase letter and a number or a special character.",
-              },
-            }}
+          <Button
+            label="Sign up with Google"
+            variant="secondary"
+            onClick={handleGoogleSignUp}
+            imgSrc={googleLogoImg}
+            imgAlt="Google Logo"
           />
-        </section>
-        <button
-          form="register-form"
-          type="submit"
-          className="bg-button-purple h-[48px] sm:h-[44px] w-full sm:w-[117px] text-basic-white mx-auto hover:brightness-[110%] transition-all delay-100 ease-in-out"
-        >
-          Sign Up
-        </button>
-      </form>
-    </FormProvider>
+        </div>
+      ) : (
+        <FormProvider {...form}>
+          <form id="register-form" onSubmit={form.handleSubmit(onSubmit)}>
+            <section className="flex flex-col form-vertical-margin form-bottom-padding">
+              <label>Nickname</label>
+              <InputField
+                name="name"
+                placeholder="Enter a nickname..."
+                type="text"
+                validation={{
+                  required: true,
+                  minLength: {
+                    value: usernameMinLength,
+                    message: `Nickname must be atleast ${usernameMinLength} characters long.`,
+                  },
+                  maxLength: {
+                    value: usernameMaxLength,
+                    message: `Nickname cannot be longer than ${usernameMaxLength} characters.`,
+                  },
+                }}
+              />
+              <label>E-mail address</label>
+              <InputField
+                name="email"
+                placeholder="Enter your email..."
+                type="email"
+                validation={{
+                  required: true,
+                  pattern: {
+                    value: emailRegexp,
+                    message: "Incorrect email format submitted.",
+                  },
+                }}
+              />
+              <label>Password</label>
+              <InputField
+                name="password"
+                placeholder="Enter a password..."
+                type="password"
+                validation={{
+                  required: true,
```

Comment:
You can extract the validation object, it will look cleaner.

Reply:
Fixed.